variables.tf

```diff
@@ -0,0 +1,224 @@
+variable "public_ip_config" {
+  type = object({
+    enabled           = bool
+    allocation_method = optional(string, "Static")
+  })
+  default = {
+    enabled = false
+  }
+  validation {
+    condition     = contains(["Static", "Dynamic"], var.public_ip_config.allocation_method)
+    error_message = "Allocation method must be Static or Dynamic"
+  }
+  description = <<-DOC
+  ```
+    enabled: Optionally select true if a public ip should be created. Defaults to false.
+    allocation_method: The allocation method of the public ip that will be created. Defaults to static.      
+  ```
+  DOC
+}
+
+# nsg needs to be an object to use the count object in main.tf. 
+variable "nic_config" {
```

Comment:
is nic a known abbreviation or should we use `network_interface`?

Reply:
I'd say yes. Also this is the same as in the windows module and I would like to avoid renaming it in only one place